src/main/java/de/craften/ui/swingmaterial/MaterialPanel.java

```diff
@@ -5,36 +5,80 @@
 import java.awt.*;
 
 /**
- * A material panel with elevation.
+ * A JPanel customized for Material components. What makes these panels special
+ * is the possibility of assigning them an elevation level. Elevation can help
+ * distinguishing elements inside a Material-based GUI, and any changes done to
+ * them result in nicely animated transitions, helping to achieve that Material
+ * flow.
+ * <p/>
+ * However, there is a catch: shadows are kinda expensive to compute, as
+ * Java2D relies on the CPU for anything other than images, so having a lot
+ * of elements with a given elevation (and thus, a shadow) can reduce
+ * performance when these elevations change because of the triggered animations.
+ * <p/>
+ * Letting the components suggest a prefered size based on their contents is
+ * still under development, so it is not advised to use your favorite
+ * {@link LayoutManager} inside a {@code MaterialPanel} unless you set the
+ * prefered, minimum and maximum size of each component by your self. Currently,
+ * the prefereable approach to follow is overriding {@link #doLayout()} and
+ * taking care of any arrangements by yourself.
  */
 public class MaterialPanel extends JPanel {
     private final ElevationEffect elevation;
 
     /**
-     * Creates a new panel.
+     * Creates a new {@code MaterialPanel}. For the moment, this is the only
+     * supported constructor from JPanel.
      */
     public MaterialPanel() {
+        super();
```

Comment:
Sorry, it's more like an habit I have. I am aware the compiler adds the call by itself, yet I always include it explicitly in source code. It helps me guiding myself when reading codes.

But this is your library, so...